### DISCUSSION TITLE
What Programming language are you going to learn in 2021?

### DISCUSSION THREAD

**Discussion by AllanRegush** [EddieHubCommunity/support#559] [General]
In [The Pragmatic Programmer](https://pragprog.com/titles/tpp20/the-pragmatic-programmer-20th-anniversary-edition/) the authors recommend learning a new language every year.

What Language(s) did you learn this year(2020)?

What Language(s) do you want to learn next year(2021)? 

For myself ->
2020: Powershell, TypeScript, Oracle SQL(Does this count?) 🤓 

2021: Rust

**Comment by mstanciu552** [comment]
2020: TypeScript, SQL, C#

2021: Kotlin

**Reply by BOLT04** [reply]
yeahhh Kotlin 👍 👍  😃

**Reply by eddiejaoude** [reply]
Yeah typescript 👍 👍 👍  (one more thumbs up than @BOLT04 😜 )

**Comment by BOLT04** [comment]
2020: nothing new
2021: Go

**Reply by eddiejaoude** [reply]
Go! Nice 👍

**Comment by vinzvinci** [comment]
2020: oop, data structures and algorithm (Java)
2021: ruby, rails🤔

**Comment by emmysteven** [comment]
**2020:** C#
**2021:** Typescript

**Comment by ghost** [comment]
2020 : Reactjs , Nodejs 
2021 : First target is Mongodb , after that Rust or go or deno (not decided yet will decide on basis of trend and market)

**Comment by Vyvy-vi** [comment]
2020: Python, Dictu, C, html
2021: html, js, front-end framework(Vue, React), some data analysis libraries for python

**Comment by The-Noah** [comment]
2020: I don't think anything new, mainly just refining my existing skills.
2021: Rust (though technically I started a few days ago 😁)

**Reply by emmysteven** [reply]
Great, I guess you are trying not to bite more than you can chew 😁

**Comment by eddiejaoude** [comment]
2020: briefly looked at NuxtJS
2021: NuxtJS, FeathersJS,  more into building Docker containers and deploying

**Reply by The-Noah** [reply]
Docker containers is something I'm thinking of looking into as well.

**Reply by eddiejaoude** [reply]
Cool, we will share what we learn 👍

**Reply by eddiejaoude** [reply]
NestJS is now on my radar, it looks more mature than featherJS

**Reply by The-Noah** [reply]
Hmm, that seems interesting. I may look into that at some point.

**Comment by mikeysan** [comment]
More Python then Flutter/Dart

**Reply by Vyvy-vi** [reply]
mikey straying from Python O_O

**Reply by eddiejaoude** [reply]
Nice Mikey 👍

**Reply by Akshu-on-github** [reply]
I'm learning dart too!

**Comment by Kitecatcher** [comment]
I want to learn whatever needed to reach my goals currently I'm a youngling at  Arduino, Python, but I need to learn HTML as well maybe. like so many others I want to work on a gpt3 style project but I don't know where to start.

**Comment by imvisp** [comment]
1.python (in depth)

**Reply by eddiejaoude** [reply]
Nice 👍

**Reply by Vyvy-vi** [reply]
🐍  python is great

**Comment by abughurab16** [comment]
2020: Python, javascript, c# , html & css (yes they count)
2021:Vala,C,MySQL and some data science tools

**Reply by eddiejaoude** [reply]
Nice 👍

**Comment by rakeshdeka** [comment]
1. **C++**
2. **Kotlin**
3. **JavaScript**

**Reply by eddiejaoude** [reply]
Cool 👍

**Comment by jcads** [comment]
This year I want to develop mobile apps, so I'll be learning Dart :)

**Reply by eddiejaoude** [reply]
Nice 👍

**Comment by RemcoHalman** [comment]
2020: Javascript/Vue, Python
2021: Typescript, python data science, Dart, CI in depth

**Reply by eddiejaoude** [reply]
Nice 👍

**Comment by DevCassie** [comment]
Not sure what language I should learn in 2021, because I'm mainly a front-end developer / graphic designer. However, I do get some C# classes for games at my college. Alongside that, I've been thinking of trying TypeScript sooner or later this year. 

I'm not sure what languages are interesting for a designer. So far I've just done HTML, CSS/SASS, JavaScript and NodeJS haha. Perhaps someone has some cool suggestions. I'm open for everything honestly. I like learning 🤓

**Comment by abughurab16** [comment]
you can check out vala and maybe try making apps for elementary OS
that sounds like something a graphic designer could dwell in?

**Reply by DevCassie** [reply]
> you can check out vala and maybe try making apps for elementary OS
> that sounds like something a graphic designer could dwell in?
> 
> On 3/2/21, Cassie <notifications@github.com> wrote:
> > Not sure what language I should learn in 2021, because I'm mainly a
> > front-end developer / graphic designer. However, I do get some C# classes
> > for games at my college. Alongside that, I've been thinking of trying
> > TypeScript sooner or later this year.
> 
> >
> 
> > I'm not sure what languages are interesting for a designer. So far I've just
> > done HTML, CSS/SASS, JavaScript and NodeJS haha. Perhaps someone has some
> > cool suggestions. I'm open for everything honestly. I like learning 🤓
> 
> >
> 
> > --
> 
> > You are receiving this because you commented.
> 
> > Reply to this email directly or view it on GitHub:
> 
> > https://github.com/EddieHubCommunity/support/discussions/559#discussioncomment-423079
> 

Thank you! I'll take a look into that.

**Comment by suubh** [comment]
2021 : Continue JavaScript.
